src/database/mod.rs

```diff
@@ -434,6 +435,65 @@ impl Database {
         Ok(identity)
     }
 
+    // TODO: Change status to enum
+    pub async fn get_unprocessed_commitments(
+        &self,
+        status: Status,
+    ) -> Result<Vec<types::UnprocessedCommitment>, Error> {
+        let query = sqlx::query(
+            r#"
+                SELECT * FROM unprocessed_identities
+                WHERE status = $1
+            "#,
+        )
+        .bind(<&str>::from(status));
+
+        let result = self.pool.fetch_all(query).await?;
+
+        Ok(result
+            .into_iter()
+            .map(|row| types::UnprocessedCommitment {
+                commitment:    row.get::<Hash, _>(0),
+                status:        row.get::<&str, _>(1).parse().expect("status is unreadable"),
+                created_at:    row.get::<_, _>(2),
+                processed_at:  row.get::<_, _>(3),
+                error_message: row.get::<_, _>(4),
+            })
+            .collect::<Vec<_>>())
+    }
+
+    pub async fn remove_unprocessed_identity(&self, commitment: &Hash) -> Result<(), Error> {
+        let query = sqlx::query(
+            r#"
+                DELETE FROM unprocessed_identities WHERE commitment = $1
+            "#,
+        )
+        .bind(commitment);
+
+        self.pool.execute(query).await?;
+
+        Ok(())
+    }
+
+    pub async fn update_err_unprocessed_commitment(
+        &self,
+        commitment: Hash,
+        message: String,
+    ) -> Result<(), Error> {
+        let query = sqlx::query(
+            r#"
+                UPDATE unprocessed_identities SET error_message = $1, status = Failed
+                WHERE commitment = $2
+            "#,
+        )
+        .bind(commitment)
+        .bind(message);
```

Comment:
Wrong order, commitment is `$2` and message is `$1`, no?

Reply:
And speaking of, could you add tests for these new methods?

Reply:
tests added